ios/brave-ios/Sources/DataImporter/UI/DataImporterView.swift

```diff
@@ -0,0 +1,361 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import BraveShared
+import DesignSystem
+import SwiftUI
+import UniformTypeIdentifiers
+import os.log
+
+private struct ImporterInfo {
+  var shouldShowFileImporter: Bool
+  var fileTypes = [UTType.zip]
+  var importType = DataImportType.all
+}
+
+public struct DataImportView: View {
+
+  @Environment(\.dismiss)
+  private var dismiss
+
+  @State
+  private var showSuccessView = false
+
+  @State
+  private var showFailureView = false
+
+  @State
+  private var showLoadingView = false
+
+  @State
+  private var showMainView = true
+
+  @State
+  private var importerInfo = ImporterInfo(shouldShowFileImporter: false)
+
+  @ObservedObject
+  private var model = DataImportModel()
+
+  private var openURL: (URL) -> Void
+  private var onDismiss: () -> Void
+
+  public init(openURL: @escaping (URL) -> Void, onDismiss: @escaping () -> Void) {
+    self.openURL = openURL
+    self.onDismiss = onDismiss
+  }
+
+  public var body: some View {
+    ZStack {
+      if model.importState == .success {
+        DataImporterStateView(kind: .success, model: model) {
```

Comment:
There's something weird here about creating an entire switch across import states all to display the same View (`DataImporterStateView`) with almost identical modifiers with the only thing changing is Buttons? You also seem to combine both making this reusable and a "View only" kind of component and also passing in the model which means you're now checking specific things related to the logic of the feature.

Either you should have one view and you only pass in the model and adjust inside directly based on `importState`, or you don't pass in the model and make sure you can pass in all necessary info from here.

You also probably want to find a way to do this without multiple if-else causing the creation of completely separate views for each state

Reply:
Switched them all to separate views.